lib/opam_build.ml

```diff
@@ -68,15 +70,14 @@ let install_project_deps ~base ~opam_files ~variant ~for_user =
   distro_extras @@
   workdir "/src" @@
   run "sudo chown opam /src" @@
+  run "cd ~/opam-repository && (git reset --hard %s || (git pull origin master && git reset --hard %s)) && opam update" commit commit @@
```

Comment:
```suggestion
  run "cd ~/opam-repository && (git reset --hard %s || (git pull origin master && git reset --hard %s)) && opam update -u" commit commit @@
```

Useful to have the update step here in case there has been a change to the compiler description files at this stage of the build.